internal/fsutil/create_test.go

```diff
@@ -62,37 +67,101 @@ var createTests = []createTest{{
 	options: fsutil.CreateOptions{
 		Path: "foo",
 		Mode: fs.ModeDir | 0775,
 	},
-	hackdir: func(c *C, dir string) {
-		c.Assert(os.Mkdir(filepath.Join(dir, "foo/"), fs.ModeDir|0765), IsNil)
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.Mkdir(filepath.Join(targetDir, "foo/"), fs.ModeDir|0765), IsNil)
 	},
 	result: map[string]string{
 		// mode is not updated.
 		"/foo/": "dir 0765",
 	},
 }, {
+	summary: "Re-creating an existing file keeps the original mode",
 	options: fsutil.CreateOptions{
 		Path: "foo",
 		// Mode should be ignored for existing entry.
 		Mode: 0644,
 		Data: bytes.NewBufferString("changed"),
 	},
-	hackdir: func(c *C, dir string) {
-		c.Assert(os.WriteFile(filepath.Join(dir, "foo"), []byte("data"), 0666), IsNil)
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.WriteFile(filepath.Join(targetDir, "foo"), []byte("data"), 0666), IsNil)
 	},
 	result: map[string]string{
 		// mode is not updated.
 		"/foo": "file 0666 d67e2e94",
 	},
+}, {
+	summary: "Create a hard link",
+	options: fsutil.CreateOptions{
+		Path:        "dir/link-to-file",
+		Link:        "file",
+		Mode:        0644,
+		MakeParents: true,
+	},
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.WriteFile(filepath.Join(targetDir, "file"), []byte("data"), 0644), IsNil)
+		// An absolute path is required to create a hard link.
+		options.Link = filepath.Join(targetDir, options.Link)
+	},
+	result: map[string]string{
+		"/file":             "file 0644 3a6eb079",
+		"/dir/":             "dir 0755",
+		"/dir/link-to-file": "file 0644 3a6eb079",
+	},
+}, {
+	summary: "Cannot create a hard link if the link target does not exist",
+	options: fsutil.CreateOptions{
+		Path:        "dir/link-to-file",
+		Link:        "file-no-exist",
+		Mode:        0644,
+		MakeParents: true,
+	},
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		options.Link = filepath.Join(targetDir, options.Link)
+	},
+	error: `link target does not exist: \/[^ ]*\/file-no-exist`,
+}, {
+	summary: "Re-creating a duplicated hard link keeps the original link",
+	options: fsutil.CreateOptions{
+		Path:        "link-to-file",
+		Link:        "file",
+		Mode:        0644,
+		MakeParents: true,
+	},
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.WriteFile(filepath.Join(targetDir, "file"), []byte("data"), 0644), IsNil)
+		c.Assert(os.Link(filepath.Join(targetDir, "file"), filepath.Join(targetDir, "link-to-file")), IsNil)
+		options.Link = filepath.Join(targetDir, options.Link)
+	},
+	result: map[string]string{
+		"/file":         "file 0644 3a6eb079",
+		"/link-to-file": "file 0644 3a6eb079",
+	},
+}, {
+	summary: "Cannot create a hard link if the link path exists and it is not a hard link to the target",
+	options: fsutil.CreateOptions{
+		Path:        "link-to-file",
```

Comment:
```suggestion
		Path:        "hardlink",
```
In general the names were becoming too long and it was not adding a lot to the readability because we don't care about the target unless it is important (missing file for example).

Reply:
Fixed in [03ce4d3](https://github.com/letFunny/chisel/pull/17/commits/03ce4d322ccaff4f6ba8c00a32aee039de12c9d3).